doc/Command-Reference.md

```diff
@@ -4119,6 +4119,18 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#Warm-R
 
 ## Warm Restart
 
+Besides device level warm reboot, SoNIC also provides docker based warm restart. This feature is currently supported by following dockers: BGP, teamD,  and SWSS. A user can manage to restart a particular docker, with no interruption on packet forwarding and no effect on other services. This helps to reduce operational costs as well as development efforts. For example, to fix a bug in BGP routing stack, only the BGP docker image needs to be built, upgraded and tested.
+
+To achieve uninterrupted packet forwarding during the restarting stage and database reconciliation at the post restarting stage, warm reboot enabled dockers with adjacency state machine facilitate standardized protocols. For example, a BGP restarting must have BGP graceful restart enabled and its BGP neighbors must be graceful restart helper capable, which are specified in [IETF RFC4724](https://tools.ietf.org/html/rfc4724). Before warm restart BGP docker, the following BGP commands should be configured:
+
+  bgp graceful-restart
+  bgp graceful-restart preserve-fw-state
```

Comment:
Is this something not enabled by default? This configuration entry is currently not in any config template. Should we add it?

Reply:
This should be added, otherwise BGP GR will fail

Reply:
This configuration is currently not in template. Do you know if the default value is enabled? If it is crucial for GR. I take it is enabled by default otherwise we would have seen failures already?

Reply:
It's enabled by default